src/provider/application.js

```diff
@@ -167,8 +192,7 @@ class Application extends EventEmitter {
   */
   launch() {
     if (window.self !== window.top) {
-      // 1: Setup listeners for all incoming communication and beforeunload
-      window.addEventListener('message', this.handleConsumerMessage);
+      // 1: Setup listener for beforeunload
       window.addEventListener('beforeunload', this.unload);
```

Comment:
i'd probably move this out to initialize too

Reply:
Moved: dab024f